homeassistant/components/sensor/solaredge.py

```diff
@@ -0,0 +1,161 @@
+"""
+Support for SolarEdge Monitoring API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.solaredge/
+"""
+
+import asyncio
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.const import CONF_MONITORED_VARIABLES
+from homeassistant.components.light import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_point_in_utc_time
+from homeassistant.util import dt as dt_util
+
+REQUIREMENTS = ['solaredge==0.0.2']
+
+DOMAIN = "solaredge"
+
+# Config for solaredge monitoring api requests.
+CONF_API_KEY = "api_key"
+CONF_SITE_ID = "site_id"
+
+DELAY_OK = 10
+DELAY_NOT_OK = 20
+
+# Supported sensor types:
+# Key: ['name', unit, icon]
+SENSOR_TYPES = {
+    'lifeTimeData': ["Lifetime energy", 'Wh', 'mdi:solar-power'],
+    'lastYearData': ["Energy this year", 'Wh', 'mdi:solar-power'],
+    'lastMonthData': ["Energy this month", 'Wh', 'mdi:solar-power'],
+    'lastDayData': ["Energy today", 'Wh', 'mdi:solar-power'],
+    'currentPower': ["Current Power", 'W', 'mdi:solar-power']
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_SITE_ID): cv.string,
+    vol.Optional(CONF_MONITORED_VARIABLES, default=[]):
+    vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)])
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_entities,
+                         discovery_info=None):
+    """Create the SolarEdge Monitoring API sensor."""
+    api_key = config.get(CONF_API_KEY, None)
+    site_id = config.get(CONF_SITE_ID, None)
+
+    # Create solaredge data service which will retrieve and update the data.
+    data = SolarEdgeData(hass, api_key, site_id)
+
+    # Create a new sensor for each sensor type.
+    entities = []
+    for sensor_type in config[CONF_MONITORED_VARIABLES]:
+        sensor = SolarEdgeSensor(sensor_type, data)
+        entities.append(sensor)
+
+    async_add_entities(entities, True)
+
+    # Schedule first data service update straight away.
+    async_track_point_in_utc_time(hass, data.async_update, dt_util.utcnow())
+
+
+class SolarEdgeSensor(Entity):
+    """Representation of an SolarEdge Monitoring API sensor."""
+
+    def __init__(self, sensorType, data):
+        """Initialize the sensor."""
+        self.type = sensorType
+        self.data = data
+
+        self._name = SENSOR_TYPES[self.type][0]
+        self._unit_of_measurement = SENSOR_TYPES[self.type][1]
+
+    @property
+    def name(self):
+        """Return the name."""
+        return 'solaredge_' + self.type
```

Comment:
Would be nice if the user could overwrite this directly with the platform configuration -> `CONF_NAME` with a default of "SolarEdge".